--- a/components/autofill/core/common/autofill_prefs.cc
+++ b/components/autofill/core/common/autofill_prefs.cc
@@ -35,7 +35,7 @@ constexpr char kAutofillRanExtraDeduplication[] =
 void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
   // Synced prefs. Used for cross-device choices, e.g., credit card Autofill.
   registry->RegisterBooleanPref(
-      kAutofillProfileEnabled, false,
+      kAutofillProfileEnabled, true,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterIntegerPref(
       kAutofillLastVersionDeduped, 0,
@@ -63,7 +63,7 @@ void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
       kAutofillHasSeenIban, false,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterBooleanPref(
-      kAutofillCreditCardEnabled, false,
+      kAutofillCreditCardEnabled, true,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterBooleanPref(
       kAutofillPaymentCvcStorage, true,
--- a/components/password_manager/core/browser/password_manager.cc
+++ b/components/password_manager/core/browser/password_manager.cc
@@ -547,10 +547,10 @@ PasswordForm CreateFormForLeakCheck(const autofill::FormData& form_data,
 void PasswordManager::RegisterProfilePrefs(
     user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterBooleanPref(
-      prefs::kCredentialsEnableService, false,
+      prefs::kCredentialsEnableService, true,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PRIORITY_PREF);
   registry->RegisterBooleanPref(
-      prefs::kCredentialsEnableAutosignin, false,
+      prefs::kCredentialsEnableAutosignin, true,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PRIORITY_PREF);
   registry->RegisterBooleanPref(
       prefs::kWasAutoSignInFirstRunExperienceShown, false,
--- a/components/payments/core/payment_prefs.cc
+++ b/components/payments/core/payment_prefs.cc
@@ -11,7 +11,7 @@ namespace payments {
 void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterBooleanPref(kPaymentsFirstTransactionCompleted, false);
   registry->RegisterBooleanPref(
-      kCanMakePaymentEnabled, false,
+      kCanMakePaymentEnabled, true,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
 }
 
--- a/chrome/browser/importer/importer_list.cc
+++ b/chrome/browser/importer/importer_list.cc
@@ -254,11 +254,6 @@ std::vector<user_data_importer::SourceProfile> DetectSourceProfiles(
     profiles.push_back(bookmarks_profile);
   }
 
-  for (auto& profile : profiles) {
-    profile.services_supported &= ~(user_data_importer::PASSWORDS
-                                    | user_data_importer::AUTOFILL_FORM_DATA);
-  }
-
   return profiles;
 }
 
@@ -298,10 +293,5 @@ void ImporterList::SourceProfilesLoaded(
 
   source_profiles_.assign(profiles.begin(), profiles.end());
 
-  for (auto& profile : source_profiles_) {
-    if (profile.services_supported & user_data_importer::PASSWORDS)
-      profile.services_supported ^= user_data_importer::PASSWORDS;
-  }
-
   std::move(profiles_loaded_callback).Run();
 }
