--- a/chrome/browser/renderer_context_menu/render_view_context_menu.cc
+++ b/chrome/browser/renderer_context_menu/render_view_context_menu.cc
@@ -1161,6 +1161,11 @@ void RenderViewContextMenu::InitMenu() {
     AppendSpellingAndSearchSuggestionItems();
   }
 
+  if (content_type_->SupportsGroup(
+          ContextMenuContentType::ITEM_GROUP_AUTOFILL)) {
+    autofill_context_menu_manager_.AppendItems();
+  }
+
   if (editable) {
     AppendOtherEditableItems();
   }
--- a/chrome/browser/ui/omnibox/omnibox_pedal_implementations.cc
+++ b/chrome/browser/ui/omnibox/omnibox_pedal_implementations.cc
@@ -2029,6 +2029,8 @@ GetPedalImplementations(bool incognito, bool guest, bool testing) {
   if (!incognito && !guest) {
     add(new OmniboxPedalClearBrowsingData(/*incognito=*/false));
   }
+  add(new OmniboxPedalManagePasswords());
+  add(new OmniboxPedalUpdateCreditCard());
   add(new OmniboxPedalLaunchIncognito());
   add(new OmniboxPedalPlayChromeDinoGame());
   add(new OmniboxPedalManageSiteSettings());
@@ -2045,10 +2047,13 @@ GetPedalImplementations(bool incognito, bool guest, bool testing) {
   if (!guest) {
     add(new OmniboxPedalClearBrowsingData(incognito));
   }
+  add(new OmniboxPedalManagePasswords());
+  add(new OmniboxPedalUpdateCreditCard());
   add(new OmniboxPedalLaunchIncognito());
   add(new OmniboxPedalUpdateChrome());
   add(new OmniboxPedalManageSecuritySettings());
   add(new OmniboxPedalManageCookies());
+  add(new OmniboxPedalManageAddresses());
   add(new OmniboxPedalManageSiteSettings());
   if (incognito) {
     add(new OmniboxPedalCloseIncognitoWindows());
--- a/chrome/browser/ui/passwords/manage_passwords_ui_controller.cc
+++ b/chrome/browser/ui/passwords/manage_passwords_ui_controller.cc
@@ -1264,10 +1264,8 @@ void ManagePasswordsUIController::UpdatePasswordIconAndBubbleState(
   }
   // Update the visibility of the page action based on the current state,
   // blocklist status, and the passwords action item.
-  if (passwords_action_item) {
-    controller->UpdateVisibility(state, is_blocklisted, *this,
-                                 *passwords_action_item);
-  }
+  controller->UpdateVisibility(state, is_blocklisted, *this,
+                               *passwords_action_item);
 
   if (show_bubble) {
     PasswordBubbleViewBase::CloseCurrentBubble();
--- a/chrome/browser/ui/toolbar/app_menu_model.cc
+++ b/chrome/browser/ui/toolbar/app_menu_model.cc
@@ -1826,6 +1826,18 @@ void AppMenuModel::Build() {
 
   AddSeparator(ui::NORMAL_SEPARATOR);
 
+  if (!browser_->profile()->IsGuestSession()) {
+    sub_menus_.push_back(
+        std::make_unique<PasswordsAndAutofillSubMenuModel>(this));
+    AddSubMenuWithStringIdAndVectorIcon(
+        this, IDC_PASSWORDS_AND_AUTOFILL_MENU,
+        IDS_PASSWORDS_AND_AUTOFILL_MENU, sub_menus_.back().get(),
+        vector_icons::kPasswordManagerIcon);
+    SetElementIdentifierAt(
+        GetIndexOfCommandId(IDC_PASSWORDS_AND_AUTOFILL_MENU).value(),
+        kPasswordAndAutofillMenuItem);
+  }
+
   if (!browser_->profile()->IsOffTheRecord()) {
     auto recent_tabs_sub_menu =
         std::make_unique<RecentTabsSubMenuModel>(provider_, browser_);
--- a/chrome/browser/ui/browser_actions.cc
+++ b/chrome/browser/ui/browser_actions.cc
@@ -848,6 +848,76 @@ void BrowserActions::InitializeBrowserActions(
           .SetVisible(!sharing_hub::SharingIsDisabledByPolicy(profile))
           .Build());
 
+  root_action_item_->AddChild(
+      ChromeMenuAction(
+          base::BindRepeating(
+              [](BrowserWindowInterface* bwi, TabStripModel* tab_strip_model,
+                 actions::ActionItem* item,
+                 actions::ActionInvocationContext context) {
+                auto page_action_trigger =
+                    context.GetProperty(page_actions::kPageActionTriggerKey);
+                // If triggered by omnibox page action, do nothing.
+                if (page_action_trigger !=
+                    page_actions::kInvalidPageActionTrigger) {
+                  return;
+                }
+
+                auto* controller = autofill::AddressBubblesIconController::Get(
+                    tab_strip_model->GetActiveWebContents());
+                if (controller && controller->GetBubbleView()) {
+                  controller->GetBubbleView()->Hide();
+                } else {
+                  chrome::ShowAddresses(bwi);
+                }
+              },
+              bwi, tab_strip_model),
+          kActionShowAddressesBubbleOrPage,
+          IDS_ADDRESSES_AND_MORE_SUBMENU_OPTION,
+          IDS_ADDRESSES_AND_MORE_SUBMENU_OPTION,
+          vector_icons::kLocationOnChromeRefreshIcon)
+          .SetEnabled(!is_guest_session)
+          .Build());
+
+  root_action_item_->AddChild(
+      ChromeMenuAction(
+          base::BindRepeating(
+              [](BrowserWindowInterface* bwi, TabStripModel* tab_strip_model,
+                 actions::ActionItem* item,
+                 actions::ActionInvocationContext context) {
+                auto page_action_trigger =
+                    context.GetProperty(page_actions::kPageActionTriggerKey);
+                // When page action is migrated, clicking on the omnibox page
+                // should not close the bubble or navigate to `Payment Methods`
+                // settings page.
+                // Page action trigger is a valid value only when this action
+                // is triggered from the migrated page action icon.
+                if (page_action_trigger !=
+                    page_actions::kInvalidPageActionTrigger) {
+                  return;
+                }
+
+                auto hide_bubble = [tab_strip_model](int command_id) -> bool {
+                  auto* controller = autofill::SavePaymentIconController::Get(
+                      tab_strip_model->GetActiveWebContents(), command_id);
+                  if (controller && controller->GetPaymentBubbleView()) {
+                    controller->GetPaymentBubbleView()->Hide();
+                    return true;
+                  }
+                  return false;
+                };
+                const bool bubble_hidden =
+                    hide_bubble(IDC_SAVE_CREDIT_CARD_FOR_PAGE) ||
+                    hide_bubble(IDC_SAVE_IBAN_FOR_PAGE);
+                if (!bubble_hidden) {
+                  chrome::ShowPaymentMethods(bwi);
+                }
+              },
+              bwi, tab_strip_model),
+          kActionShowPaymentsBubbleOrPage, IDS_PAYMENT_METHOD_SUBMENU_OPTION,
+          IDS_PAYMENT_METHOD_SUBMENU_OPTION, kCreditCardChromeRefreshIcon)
+          .SetEnabled(!is_guest_session)
+          .Build());
+
   // TODO(crbug.com/435220196): Ideally this action would have
   // ChromeLabsCoordinator passed in as a dependency directly.
   if (IsChromeLabsEnabled() && !web_app::AppBrowserController::IsWebApp(bwi)) {
@@ -935,6 +999,35 @@ void BrowserActions::InitializeBrowserActions(
   }
 
   root_action_item_->AddChild(
+      ChromeMenuAction(
+          base::BindRepeating(
+              [](BrowserWindowInterface* bwi, TabStripModel* tab_strip_model,
+                 actions::ActionItem* item,
+                 actions::ActionInvocationContext context) {
+                content::WebContents* const web_contents =
+                    tab_strip_model->GetActiveWebContents();
+                if (PasswordsModelDelegateFromWebContents(web_contents)
+                        ->GetState() == password_manager::ui::INACTIVE_STATE) {
+                  chrome::ShowPasswordManager(bwi);
+                } else {
+                  auto* const controller =
+                      ManagePasswordsUIController::FromWebContents(
+                          web_contents);
+                  if (controller->IsShowingBubble()) {
+                    controller->HideBubble(
+                        /*initiated_by_bubble_manager=*/false);
+                  } else {
+                    chrome::ManagePasswordsForPage(bwi);
+                  }
+                }
+              },
+              bwi, tab_strip_model),
+          kActionShowPasswordsBubbleOrPage, IDS_VIEW_PASSWORDS,
+          IDS_VIEW_PASSWORDS, vector_icons::kPasswordManagerIcon)
+          .SetEnabled(!is_guest_session)
+          .Build());
+
+  root_action_item_->AddChild(
       actions::ActionItem::Builder(
           base::BindRepeating(
               [](TabStripModel* tab_strip_model, actions::ActionItem* item,
--- a/chrome/browser/ui/views/location_bar/location_bar_view.cc
+++ b/chrome/browser/ui/views/location_bar/location_bar_view.cc
@@ -505,2 +505,4 @@
     params.types_enabled.push_back(PageActionIconType::kClickToCall);
+    params.types_enabled.push_back(PageActionIconType::kAutofillAddress);
+    params.types_enabled.push_back(PageActionIconType::kManagePasswords);
     if (!apps::features::ShouldShowLinkCapturingUX()) {
@@ -510,9 +512,18 @@
     params.types_enabled.push_back(PageActionIconType::kTranslate);
     params.types_enabled.push_back(PageActionIconType::kFileSystemAccess);
 
+    params.types_enabled.push_back(PageActionIconType::kCookieControls);
+    params.types_enabled.push_back(
+        PageActionIconType::kPaymentsOfferNotification);
     params.types_enabled.push_back(PageActionIconType::kMemorySaver);
   }
 
+  params.types_enabled.push_back(PageActionIconType::kSaveCard);
+  params.types_enabled.push_back(PageActionIconType::kSaveIban);
+  params.types_enabled.push_back(PageActionIconType::kFilledCardInformation);
+  params.types_enabled.push_back(PageActionIconType::kVirtualCardEnroll);
+  params.types_enabled.push_back(PageActionIconType::kMandatoryReauth);
+
   if (browser_ && !is_popup_mode_) {
     params.types_enabled.push_back(PageActionIconType::kBookmarkStar);
   }
--- a/chrome/browser/ui/web_applications/app_browser_controller.cc
+++ b/chrome/browser/ui/web_applications/app_browser_controller.cc
@@ -408,6 +408,7 @@
 
   std::vector<actions::ActionId> types_enabled = {
       kActionFind,
+      kActionShowPasswordsBubbleOrPage,
       kActionZoomNormal,
       kActionShowFileSystemAccess,
       kActionShowCookieControls,
@@ -432,7 +433,9 @@
 
   std::vector<PageActionIconType> types_enabled;
   types_enabled.push_back(PageActionIconType::kFind);
+  types_enabled.push_back(PageActionIconType::kManagePasswords);
   types_enabled.push_back(PageActionIconType::kFileSystemAccess);
+  types_enabled.push_back(PageActionIconType::kSaveCard);
 
   return types_enabled;
 }
--- a/chrome/browser/ui/webui/side_panel/customize_chrome/customize_toolbar/customize_toolbar_handler.cc
+++ b/chrome/browser/ui/webui/side_panel/customize_chrome/customize_toolbar/customize_toolbar_handler.cc
@@ -51,2 +51,9 @@
       return side_panel::customize_chrome::mojom::ActionId::kNewIncognitoWindow;
+    case kActionShowPasswordsBubbleOrPage:
+      return side_panel::customize_chrome::mojom::ActionId::
+          kShowPasswordManager;
+    case kActionShowPaymentsBubbleOrPage:
+      return side_panel::customize_chrome::mojom::ActionId::kShowPaymentMethods;
+    case kActionShowAddressesBubbleOrPage:
+      return side_panel::customize_chrome::mojom::ActionId::kShowAddresses;
     case kActionShowDownloads:
@@ -97,2 +104,8 @@
       return kActionNewIncognitoWindow;
+    case side_panel::customize_chrome::mojom::ActionId::kShowPasswordManager:
+      return kActionShowPasswordsBubbleOrPage;
+    case side_panel::customize_chrome::mojom::ActionId::kShowPaymentMethods:
+      return kActionShowPaymentsBubbleOrPage;
+    case side_panel::customize_chrome::mojom::ActionId::kShowAddresses:
+      return kActionShowAddressesBubbleOrPage;
     case side_panel::customize_chrome::mojom::ActionId::kShowDownloads:
@@ -303,7 +316,13 @@
   add_action(kActionNewIncognitoWindow,
              side_panel::customize_chrome::mojom::CategoryId::kNavigation);
 
+  add_action(kActionShowPasswordsBubbleOrPage,
+             side_panel::customize_chrome::mojom::CategoryId::kYourChrome);
+  add_action(kActionShowPaymentsBubbleOrPage,
+             side_panel::customize_chrome::mojom::CategoryId::kYourChrome);
+  add_action(kActionShowAddressesBubbleOrPage,
+             side_panel::customize_chrome::mojom::CategoryId::kYourChrome);
   add_action(kActionSidePanelShowBookmarks,
              side_panel::customize_chrome::mojom::CategoryId::kYourChrome);
   add_action(kActionSidePanelShowHistoryCluster,
              side_panel::customize_chrome::mojom::CategoryId::kYourChrome);
