--- a/chrome/browser/ui/views/tabs/vertical/root_tab_collection_node.cc
+++ b/chrome/browser/ui/views/tabs/vertical/root_tab_collection_node.cc
@@ -175,7 +175,10 @@ void RootTabCollectionNode::OnTabStripModelChanged(
     if (selection.new_tab) {
       changed_tabs.insert(selection.new_tab);
     }
-    on_active_tab_changed_callback_list_.Notify(selection.new_tab);
+    // Creating a new tab should not force the vertical tab strip to scroll.
+    if (change.type() != TabStripModelChange::kInserted) {
+      on_active_tab_changed_callback_list_.Notify(selection.new_tab);
+    }
   }
 
   if (selection.selection_changed()) {
